lib/ex_machina/strategy.ex

```diff
@@ -66,16 +94,39 @@ defmodule ExMachina.Strategy do
             )
           end
 
-          def unquote(function_name)(factory_name, attrs \\ %{}) do
+          def unquote(function_name)(factory_name, attrs, opts) do
+            record = ExMachina.build(__MODULE__, factory_name, attrs)
+
+            unquote(function_name)(record, opts)
+          end
+
+          def unquote(function_name)(factory_name, attrs) do
             record = ExMachina.build(__MODULE__, factory_name, attrs)
 
             unquote(function_name)(record)
           end
 
+          def unquote(function_name)(factory_name) do
```

Comment:
We now handle each case directly in a separate function (instead of having optional `attrs \\ %{}`) because having the optional args conflicts with other versions of `insert/2` and `insert/3`.